### PR TITLE
Re-announcing new blocks 

### PR DESCRIPTION
**Pull request by kushti**
In this PR, the node is re-announcing new blocks received.

### CODE REVIEW

File: src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala [hunk 1]
```diff
@@ -77,6 +82,130 @@ class ErgoNodeViewSynchronizer(networkControllerRef: ActorRef,
     networkControllerRef ! SendToNetwork(msg, SendToRandom)
   }
 
+  /**
+    * Helper method which is deciding whether chain is likely nearly of fully synchronized with the network
```
Comment (greenhat) [inline src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala:86]:
```suggestion
    * Helper method which is deciding whether chain is likely nearly or fully synchronized with the network
```

**Review by greenhat — APPROVED** [review]

**Review by oskin1 — APPROVED** [review]

File: src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala [hunk 2]
```diff
@@ -77,6 +82,130 @@ class ErgoNodeViewSynchronizer(networkControllerRef: ActorRef,
...
+    */
+  private def validateAndSetStatus(remote: ConnectedPeer, pmod: ErgoPersistentModifier): Boolean = {
+    historyReaderOpt match {
+      case Some(hr) =>
+        hr.applicableTry(pmod) match {
+          case Failure(e) if e.isInstanceOf[MalformedModifierError] =>
+            log.warn(s"Modifier ${pmod.encodedId} is permanently invalid", e)
+            deliveryTracker.setInvalid(pmod.id)
+            penalizeMisbehavingPeer(remote)
+            false
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala:197]: What if it is Failure with different type of exception?

File: src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala [hunk 3]
```diff
@@ -77,6 +82,130 @@ class ErgoNodeViewSynchronizer(networkControllerRef: ActorRef,
...
+          case _ =>
+            deliveryTracker.setReceived(pmod.id, remote)
+            true
+        }
+      case None =>
+        log.error("Got modifier while history reader is not ready")
+        deliveryTracker.setReceived(pmod.id, remote)
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala:204]: So in this case it got into Received state without being checked by applicableTry? So the `received` set may contain mods which are not applicable.

File: src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala [hunk 4]
```diff
@@ -77,6 +82,130 @@ class ErgoNodeViewSynchronizer(networkControllerRef: ActorRef,
...
+        val parsed: Iterable[ErgoPersistentModifier] = parseModifiers(requestedModifiers, serializer, remote)
+        val valid = parsed.filter(validateAndSetStatus(remote, _))
+        if (valid.nonEmpty) viewHolderRef ! ModifiersFromRemote[ErgoPersistentModifier](valid)
+
+        // If chain is synced or almost synced, announce new modifiers received.
+        // Helping to push new blocks around the network faster.
+        if (chainAlmostDownloaded) {
+          val toAnnounce = valid.map(_.id).toSeq
+          log.info(s"Announcing recent modifiers: ${toAnnounce.size}, type: $typeId}")
+          val msg = Message(invSpec, Right(InvData(typeId, toAnnounce)), None)
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala:128]: I wonder, why the source is None?

Comment (kushti) [reply src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala:128]: None means there's no external source for the message (so it is sent from the node).

**Pull request merged**